receiver/awsnvmereceiver/scraper.go

```diff
@@ -224,21 +184,38 @@ func (s *nvmeScraper) getNVMeDevicesByController() (map[int]*nvmeDevices, error)
 
 func newScraper(cfg *Config,
 	settings receiver.Settings,
-	nvme nvme.DeviceInfoProvider,
+	nvmeInfo nvme.DeviceInfoProvider,
 	allowedDevices collections.Set[string],
 ) *nvmeScraper {
 	scraper := &nvmeScraper{
 		logger:         settings.TelemetrySettings.Logger,
 		mb:             metadata.NewMetricsBuilder(cfg.MetricsBuilderConfig, settings),
-		nvme:           nvme,
+		nvme:           nvmeInfo,
 		allowedDevices: allowedDevices,
 	}
 
-	scraper.collectEbs, scraper.collectInstanceStore = computeCollectFlags(cfg)
+	allScrapers := []nvme.DeviceTypeScraper{
+		ebs.NewScraper(),
+		instancestore.NewScraper(),
+	}
+
+	var enabledScrapers []nvme.DeviceTypeScraper
+	for _, ts := range allScrapers {
+		if ts.IsEnabled(&cfg.MetricsBuilderConfig.Metrics) {
+			enabledScrapers = append(enabledScrapers, ts)
+		}
+	}
+	scraper.enabledScrapers = enabledScrapers
+
+	scraper.scrapersByModel = make(map[string]nvme.DeviceTypeScraper, len(enabledScrapers))
+	for _, ts := range enabledScrapers {
+		scraper.scrapersByModel[ts.Model()] = ts
+	}
```

Comment:
Why do we need the slice? Why not create the map from the beginning?

```suggestion
	scraper.enabledScrapers = make(map[string]nvme.DeviceTypeScraper)
	for _, ts := range allScrapers {
		if ts.IsEnabled(&cfg.MetricsBuilderConfig.Metrics) {
			scraper.enabledScrapers[ts.Model()] = ts
		}
	}
```